### DISCUSSION TITLE
Trouble Building

### DISCUSSION THREAD

**Discussion by 2XD** [gucci-on-fleek/lockdown-browser#57] [Q&A] [answered]
Having problem actually building the file. I installed visual studio 2022.
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/81333771/1ee57020-11c4-47b2-9953-323d786908e1)

**Comment by 2XD** [comment]
I tried it using BuildTools as well, and got the same error. 
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/81333771/fea2c5d3-d664-42da-9986-dc2f5f9304a3)

**Reply by gucci-on-fleek** [reply]
> ```log
> & : The term 'cmd' is not recognized as the name of a cmdlet, function, script file, or operable program.
> ```

This suggests that your `PATH` got messed up somehow. What output do you get when you run the following two commands?

```powershell
echo $env:PATH
cmd.exe /?
```

**Reply by 2XD** [reply]
this is what shows up if I put that in 
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/81333771/9522e984-7136-4228-ba02-e7e6b996d769)
I appreciate you helping me btw, means a lot

**Reply by gucci-on-fleek** [reply]
That's really really strange. Your `PATH` looks correct to me, but it can't find `cmd.exe` for some reason. What do you get if you run

```powershell
C:\Windows\System32\cmd.exe
```

What about if you open the “Run” dialog by pressing <kbd>Win</kbd>+<kbd>R</kbd>, then type in `cmd` and hit enter?

**Reply by 2XD** [reply]
This is what shows up if I run it in command prompt 
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/81333771/c15568b8-9b2f-4d49-a7f0-f3abe9bf27fe)

And this is what shows up when I run it through the run dialog
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/81333771/581a7e58-0483-4d72-8f76-71c31ae0b511)

**Reply by gucci-on-fleek** [reply]
Huh, so it looks like `cmd.exe` disappeared somehow. I'm actually kinda surprised that your computer is working at all.  It should be fairly easy to fix though. In an “Administrator” Powershell window, run

```powershell
sfc /scannow
dism /online /cleanup-image /restorehealth
sfc /scannow
```

It'll take anywhere from 5 minutes to 5 hours for those to finish running, but once it's done, reboot your computer and it should hopefully be fixed.

**Reply by 2XD** [reply]
Yea ur right it actually did LOL. I typed into my windows search bar and its just a blank file. I'll keep you updated, thanks a ton!

**Reply by gucci-on-fleek** [reply]
That's really bizarre. Even if you're running as Admin, you can't delete/modify files in `C:\Windows` without trying _really_ hard, so it's not like some poorly written installer/uninstaller deleted it accidentally. A Windows Update bug maybe? One-in-a-million odds, but someone's bound to get unlucky.

(And if something goes really wrong while trying to fix this, there should be an alternate version at `C:\Windows\SysWOW64\cmd.exe` that you can run for recovery purposes.)

**Reply by 2XD** [reply, marked as answer]
After running just one sfc/scannow I got it working, and cmd was recovered! Was able to launch it and everything runs flawlessly. Thanks a ton for your help, and I also wanted to let you know I was doing it on Windows 11 Pro version so your code works for both Windows 10 and 11! Thanks again!
